examples/custom_solvers.jl

```diff
@@ -43,11 +43,40 @@ function my_eig_solver(A, X0; maxiter, tol, kwargs...)
     (λ=λ, X=X, residual_norms=[], iterations=0, converged=true, n_matvec=0)
 end;
 
+# Finally we also define our custom mixing scheme. It will be a mixture
+# of simple mixing (for the first 2 steps) and than default to Kerker mixing.
+struct MyMixing
+    n_initial
+    mixing_initial
+    mixing_default
+end
+
+MyMixing() = MyMixing(2, SimpleMixing(α=0.7), KerkerMixing(α=0.7))
+function DFTK.mix(params::MyMixing; n_iter, kwargs...)
+    mixing = n_iter > params.n_initial ? mixing_initial : mixing_default
+    mix(mixing; n_iter=n_iter, kwargs...)
+end;
+
+function mix(mix::KerkerMixing; basis, ρin::RealFourierArray, ρout::RealFourierArray, kwargs...)
+    T = eltype(basis)
+    Gsq = [sum(abs2, basis.model.recip_lattice * G)
+           for G in G_vectors(basis)]
+    ρin = ρin.fourier
+    ρout = ρout.fourier
+    ρnext = @. ρin + T(mix.α) * (ρout - ρin) * Gsq / (T(mix.kF)^2 + Gsq)
+    # take the correct DC component from ρout; otherwise the DC component never gets updated
+    ρnext[1] = ρout[1]
+    from_fourier(basis, ρnext; assume_real=true)
+end
```

Comment:
Hm, it's annoying and a bit technical to copy/paste this here. Maybe just do a trivial mixing (eg simple mixing with a coefficient that changes from iteration to iteration) to demonstrate, and then refer to mixing.jl for details on advanced mixing

Reply:
Ups that's entirely by accident and not needed.

Reply:
well then again it's maybe interesting to show how to do mixing by hand. So maybe you can do the simple mixing by hand and call kerker otherwise?